modules/programs/gh.nix

```diff
@@ -0,0 +1,93 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.gh;
+
+  package = pkgs.gitAndTools.gh;
+
+in {
+  meta.maintainers = [ maintainers.gerschtli ];
+
+  options.programs.gh = {
+    enable = mkEnableOption "GitHub CLI tool";
+
+    enableBashIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
+
+    enableZshIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+
+    enableFishIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Fish integration.
+      '';
+    };
+
+    aliases = mkOption {
+      default = { };
+      type = with types; attrsOf str;
+      example = literalExample ''
+        {
+          co = "pr checkout";
+          pv = "pr view";
+        }
+      '';
+      description = ''
+        Aliases allow you to create nicknames for gh commands.
+      '';
+    };
+
+    editor = mkOption {
+      default = "";
+      type = types.str;
+      description = ''
+        What editor gh should run when creating issues, pull
+        requests, etc. If blank, will refer to environment.
+      '';
+    };
+
+    gitProtocol = mkOption {
+      default = "https";
+      type = types.enum [ "https" "ssh" ];
+      description = ''
+        What protocol to use when performing git operations.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ package ];
+
+    programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
+      eval "$(${package}/bin/gh completion --shell bash)"
+    '';
+
+    programs.zsh.initExtra = mkIf cfg.enableZshIntegration ''
+      eval "$(${package}/bin/gh completion --shell zsh)"
+    '';
+
+    programs.fish.shellInit = mkIf cfg.enableFishIntegration ''
```

Comment:
I don't think that's necessary, the package already installs those completion files (I have autocompletion just by virtue of installing gh).

Reply:
Yes, you are right. Thank you